mobile_packet_verifier/src/daemon.rs

```diff
@@ -93,20 +97,41 @@ where
                     }
                 }
                 file = self.reports.recv() => {
-                    let Some(file) = file else {
-                        anyhow::bail!("FileInfoPoller sender was dropped unexpectedly");
+                    let Some(file_info_stream) = file else {
+                        anyhow::bail!("data transfer FileInfoPoller sender was dropped unexpectedly");
                     };
-                    tracing::info!("Verifying file: {}", file.file_info);
-                    let ts = file.file_info.timestamp;
-                    let mut transaction = self.pool.begin().await?;
-                    let reports = file.into_stream(&mut transaction).await?;
-                    crate::accumulate::accumulate_sessions(&self.mobile_config_resolver, &mut transaction, &self.verified_data_session_report_sink, ts, reports).await?;
-                    transaction.commit().await?;
-                    self.verified_data_session_report_sink.commit().await?;
+                    self.handle_data_transfer_session_file(file_info_stream).await?;
                 }
+
             }
         }
     }
+
+    async fn handle_data_transfer_session_file(
+        &self,
+        file: FileInfoStream<DataTransferSessionIngestReport>,
+    ) -> anyhow::Result<()> {
+        tracing::info!("Verifying file: {}", file.file_info);
+        let ts = file.file_info.timestamp;
+        let mut transaction = self.pool.begin().await?;
+
+        let banned_radios = banning::get_banned_radios(&mut transaction, ts).await?;
+        let reports = file.into_stream(&mut transaction).await?;
+
+        crate::accumulate::accumulate_sessions(
+            &self.mobile_config_resolver,
+            banned_radios,
+            &mut transaction,
+            &self.verified_data_session_report_sink,
+            ts,
+            reports,
+        )
+        .await?;
+
+        transaction.commit().await?;
+        self.verified_data_session_report_sink.commit().await?;
```

Comment:
What if `transaction.commit().await?;` will be committed successfully but `self.verified_data_session_report_sink.commit()` failed?

Reply:
`transaction.commit()` succeeds
- we write data transfer requests to our database for burning
- we output the record of that into the `verified_data_session_report_sink`
- and we mark the file as processed

if `self.verified_data_session_report_sink.commit()` fails
- our data is already written to the tmp file
- the file is potentially note rolled up and sent to s3
- It's picked back up on restart and uploaded